File: UsingProcessLifecycleOwnerForLibs/app/src/main/java/github/nisrulz/sample/usingprocesslifecycleownerforlibs/Main2Activity.java
package github.nisrulz.sample.usingprocesslifecycleownerforlibs;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.view.View.OnClickListener;

public class Main2Activity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        findViewById(R.id.txt_2).setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(final View v) {
                startActivity(new Intent(Main2Activity.this, MainActivity.class));
            }
        });
    }
}

File: UsingProcessLifecycleOwnerForLibs/app/src/main/java/github/nisrulz/example/usingprocesslifecycleownerforlibs/Main2Activity.kt
package github.nisrulz.example.usingprocesslifecycleownerforlibs

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import github.nisrulz.example.usingprocesslifecycleownerforlibs.databinding.ActivityMain2Binding

class Main2Activity : AppCompatActivity() {
    private lateinit var binding: ActivityMain2Binding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMain2Binding.inflate(layoutInflater)
        binding.apply {
            setContentView(root)
            txt2.setOnClickListener {
                startActivity(
                    Intent(
                        this@Main2Activity,
                        MainActivity::class.java
                    )
                )
            }
        }
    }
}